Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/Interception/AppStateListenerTests.swift

```diff
@@ -116,87 +116,98 @@ class AppStateHistoryTests: XCTestCase {
 
-        let expected = AppStateHistory(
-            initialState: .init(isActive: true, date: startDate),
-            changes: [
-                .init(isActive: false, date: startDate + 1.0),
-                .init(isActive: true, date: startDate + 2.0)
-            ],
-            recentDate: startDate + 3.0
+        // Then
+        XCTAssertEqual(
+            listener.history.currentSnapshot.state,
+            randomNotification.expectedState,
+            "It must record \(randomNotification.expectedState) after receiving '\(randomNotification.name)'"
         )
-        XCTAssertEqual(listener.history, expected)
     }
 
-    func testWhenAppBecomeActiveAndResignActive_thenAppStateHistoryIsRecorded() {
-        let startDate = Date.mockDecember15th2019At10AMUTC()
+    // MARK: - Recording History
+
+    func testWhenAppStatesHistoryIsRetrieved_thenCurrentSnapshotIsSignedWithRecentDate() {
         let listener = AppStateListener(
-            dateProvider: RelativeDateProvider(startingFrom: startDate, advancingBySeconds: 1.0),
+            dateProvider: RelativeDateProvider(startingFrom: .mockRandomInThePast(), advancingBySeconds: 1.0),
+            initialAppState: .mockRandom(),
             notificationCenter: notificationCenter
         )
-        notificationCenter.post(name: UIApplication.didBecomeActiveNotification, object: nil)
-        notificationCenter.post(name: UIApplication.willResignActiveNotification, object: nil)
-
-        let expected = AppStateHistory(
-            initialState: .init(isActive: true, date: startDate),
-            changes: [
-                .init(isActive: true, date: startDate + 1.0),
-                .init(isActive: false, date: startDate + 2.0)
-            ],
-            recentDate: startDate + 3.0
-        )
-        XCTAssertEqual(listener.history, expected)
+        let history1 = listener.history
+        let history2 = listener.history
+
+        XCTAssertEqual(history2.currentSnapshot.date.timeIntervalSince(history1.currentSnapshot.date), 1.0)
     }
 
-    func testWhenAppStateHistoryIsRetrieved_thenFinalDateOfHistoryChanges() {
-        let startDate = Date.mockDecember15th2019At10AMUTC()
+    func testWhenReceivingStateChangeNotifications_itRecordsHistoryOfAppStates() {
+        var expectedHistoryStates: [AppState] = []
+
+        // Given
         let listener = AppStateListener(
-            dateProvider: RelativeDateProvider(startingFrom: startDate, advancingBySeconds: 1.0),
+            dateProvider: RelativeDateProvider(startingFrom: .mockRandomInThePast(), advancingBySeconds: 1.0),
+            initialAppState: .mockRandom(),
             notificationCenter: notificationCenter
         )
-        let history1 = listener.history
-        let history2 = listener.history
 
-        XCTAssertEqual(history2.currentState.date.timeIntervalSince(history1.currentState.date), 1.0)
+        // When
+        (0..<1).forEach { _ in
```

Comment:
This will do a single iteration :)
```suggestion
        (0..<10).forEach { _ in
```

Reply:
This one need to be addressed before merging. All good otherwise!

Reply:
Ops! Fixed, good catch 💪